### PR TITLE
add node-env-file to hubot to parse .env files

### PR DESCRIPTION
**Pull request by twellspring**
Adding node-env-file to bin/hubot so it can read standard node .env files 
Reference:  issue #1153 

If someone can suggest where this should be documented I will add documentation changes to pull request.

### CODE REVIEW

**Comment by gasolin** [comment]
maybe it's better to also ignore .env file on `.gitignore`, so people wont check it in accidentally?

**Comment by dnrce** [comment]
What's your reasoning for choosing `node-env-file` (v0.1.8, 9246 downloads this month) over `dotenv` (v2.0.0, 249627 downloads this month)?

**Comment by twellspring** [comment]
Yes using dotenv is a better choice.  I have updated it to use dotenv.   Functionality is the same.

**Comment by bhuga** [comment]
> maybe it's better to also ignore .env file on .gitignore, so people wont check it in accidentally?

I like this thinking.

> If someone can suggest where this should be documented I will add documentation changes to pull request.

I think adding a quick line or to to the deployment platforms that this applies to is the right call.

If we want to make `dotenv` a first-class citizen, we should probably also generate it when setting up a new hubot.

**Comment by Ryangr0** [comment]
Does this have any estimated merge date? I use this already on every implementation of hubot and manually edit the hubot file now, but that's obviously a very dirty way of doing this.

**Comment by jd-carroll** [comment]
What's the status of getting this merged?

**Comment by technicalpickles** [comment]
I'm not sure I like including this as a first-class citizen for hubot, just because people are going to manage environments their own way. I don't know what side effect this would have on existing users that manage their own stuff.

I think in the long run, I'd prefer to see movement towards programatic launching of hubot ie https://github.com/github/hubot/issues/858 , and generating a new hubot could generate the code to initialize a new hubot, with a one-liner to load with dotenv.

**Comment by davidyell** [comment]
So this requires an extra dependancy?

File: bin/hubot [hunk 1]
```diff
@@ -7,6 +7,7 @@ Hubot    = require '..'
 Fs       = require 'fs'
 OptParse = require 'optparse'
 Path     = require 'path'
+env      = require('dotenv').config({silent: true});
```
Comment (gr2m) [inline bin/hubot]: you don’t need to set the `env` variable. I would also suggest to move up `require('dotenv').config({silent: true});` to the very first code line, as suggested in https://www.npmjs.com/package/dotenv#usage

**Comment by gr2m** [comment]
sorry I missed the whole discussion before my review. I agree that if we should ever agree on merging this, we should add `.env` to the generated `.gitignore` and document how to sent ENV variables. 

I still have to review #858

**Pull request closed without merging**